src/utils.h

```diff
@@ -122,6 +122,19 @@ memdup(const void *mem, size_t nmemb, size_t size)
     return p;
 }
 
+#ifdef _MSC_VER
```

Comment:
Let's use a configure check here instead of checking for MSVC, this way it'd work for any platform which might be missing `strndup`.